### PR TITLE
Add Dialyzer to Travis run

### PR DESCRIPTION
**Pull request by Baradoy**
Add Dialyzer to Travis
Fix a few Dialyzer issues
Upgrades Elixir and Dialyzer versions

Supersedes #181

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -1,2 +1,2 @@
-erlang 21.0
-elixir 1.7.1
+erlang 21.2
```
Comment (hez) [inline .tool-versions]: should this be 21.3 as well?

File: .gitignore [hunk 2]
```diff
@@ -22,3 +22,7 @@ erl_crash.dump
 
 # Ignore package tarball (built via "mix hex.build").
 shopify_api-*.tar
+
+# Dialyzer will create PLT files here 
+/priv/plts/*.plt
+/priv/plts/*.plt.hash
```
Comment (hez) [inline .gitignore]: could we just ignore all of /priv/plts ?

File: .tool-versions (hunk 1 above)
Comment (Baradoy) [reply .tool-versions]: I mean, probably....

File: .gitignore (hunk 2 above)
Comment (Baradoy) [reply .gitignore]: I think we may want to add the folder /priv/plts/ and just not the files. That being said, I don't think that is what I have done.

**Pull request merged**